superset/models/core.py

```diff
@@ -222,6 +223,17 @@ def unique_name(self) -> str:
     def url_object(self) -> URL:
         return make_url(self.sqlalchemy_uri_decrypted)
 
+    @property
+    def parameters(self) -> Optional[Dict[str, Any]]:
+        # Build parameters if db_engine_spec is a subclass of BasicParametersMixin
+        parameters = {"engine": self.backend}
+
+        if issubclass(self.db_engine_spec, BasicParametersMixin):
```

Comment:
After we add BQ we need to change this, right?

Reply:
Yea that PR will have the updated logic here to allow us to pass Big Query credentials